api-scripts/patron/placing-holds/README.md

```diff
@@ -0,0 +1,99 @@
+    "requesterId": "e96618a9-04ee-4fea-aa60-306a8f4dd89b",
+    "requestType": "Hold",
+    "fulfilmentPreference": "Hold Shelf",
+    "itemId": "0000023d-a95f-4396-8d37-42dda48cdf3d",
+    "requestExpirationDate": "2020-06-18T09:23:11.957Z",
+    "pickupServicePointId": "d80a641e-50b9-4bd2-99a2-a4a12c8515cb"
+  }`
+
+
+## Modules required to be enabled as a pre-requisite to run JMeter script:
+##### Backend:
+- mod-patron-4.1.0
+- mod-circulation-18.0.9
+- mod-circulation-storage-11.0.0
+- mod-inventory-storage-19.1.2
+- mod-inventory-14.1.3
+- mod-authtoken-2.4.0
+- mod-permissions-5.9.0
+- okapi-2.38.0
+
+- FOLIO release: FameFlower
+
+#JMX scripts 
+
+- patron_placeHolds.jmx - Place an instance-level request to mod-patron
+- patron_placeHolds_circulation_instance_level.jmx - Place an instance-level request to mod-circulation
+- patron_placeHolds_circultion_item_level.jmx - Place an item-level request to mod-circulation
+
+#Sql queries from jmeter-supported-data folder
+The ids of users and the ids of instances have been pulled which don't have requests in order to not clean requests which have already existed
+
+- user_ids_without_requests.csv (the users' ids have been cut to 10000 records, that's enough to run a 30-min test with 20 users)<br/> 
+`SELECT us.jsonb->>'id' FROM fs09000000_mod_users.users as us WHERE us.jsonb->>'id' NOT IN (SELECT req.jsonb->>'requesterId'
+	FROM fs09000000_mod_circulation_storage.request as req GROUP BY jsonb->>'requesterId') LIMIT 10000;
+`
+
+- instances_without_requests.csv (the instances' ids have been cut to 10000 records, that's enough to run a 30-min test with 20 users)<br/>
+`SELECT distinct inst.id
+	FROM fs09000000_mod_inventory_storage.instance as inst, fs09000000_mod_inventory_storage.holdings_record as hr,
+	fs09000000_mod_inventory_storage.item as it WHERE inst.id=hr.instanceid AND hr.id=it.holdingsrecordid
+	AND it.jsonb->>'id' NOT IN (SELECT req.jsonb->>'itemId'
+	FROM fs09000000_mod_circulation_storage.request as req GROUP BY jsonb->>'itemId') LIMIT 10000;
+`
+
+- items_without_requests.csv (the items' ids have been cut to 10000 records, that's enough to run a 30-min test with 20 users)<br/>
+`SELECT it.id
+ 	FROM  fs09000000_mod_inventory_storage.instance as inst, fs09000000_mod_inventory_storage.holdings_record as hr,
+ 	fs09000000_mod_inventory_storage.item as it WHERE inst.id=hr.instanceid AND hr.id=it.holdingsrecordid 
+ 	AND it.jsonb->>'id' NOT IN  (SELECT req.jsonb->>'itemId'
+ 	FROM fs09000000_mod_circulation_storage.request as req GROUP BY jsonb->>'itemId') LIMIT 10000;
+`
+
+- service_points.csv <br/>
+they are received on UI
+
+#Sql query from script folder
+- delete_requests.sql (it is a sql script to clean the requests by users' ids)
+
+## Thread groups:
+The test contains of 2 thread groups:
+- Login - for identification and authentication, it is executed only 1 time
+- Place an instance-level request - the api test which is tested 
+
+## Before the test execution:
```

Comment:
```suggestion
## After the test execution:
```
Should it be "After" the test execution because we want to restore the database so that whoever runs a test afterward will always work with a clean database?

Also, was the checkin-checkout-db-restore.sql script needed? It would truncate all the requests and loans. If it was used, did you mean that only the query to update all items to 'Available' (their original state) from checkin-checkout-db-restore.sql was used?  If the entire checkin-checkout-db-restore.sql was used, then do we need the script here: delete_requests.sql?

Reply:
Changed